src/main/java/net/imagej/ops/image/distancetransform/DefaultDistanceTransform.java

```diff
@@ -0,0 +1,349 @@
+
+import net.imagej.ops.Contingent;
+import net.imagej.ops.Op;
+import net.imagej.ops.Ops.Image.DistanceTransform;
+import net.imagej.ops.create.img.CreateImgFromDimsAndType;
+import net.imagej.ops.special.function.Functions;
+import net.imagej.ops.special.function.UnaryFunctionOp;
+import net.imagej.ops.special.hybrid.AbstractUnaryHybridCF;
+import net.imglib2.FinalInterval;
+import net.imglib2.RandomAccess;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.algorithm.labeling.ConnectedComponents.StructuringElement;
+import net.imglib2.type.BooleanType;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.type.numeric.real.FloatType;
+
+/**
+ * @author Simon Schmid (University of Konstanz)
+ */
+@Plugin(type = Op.class)
+public class DefaultDistanceTransform<B extends BooleanType<B>, T extends RealType<T>>
+		extends AbstractUnaryHybridCF<RandomAccessibleInterval<B>, RandomAccessibleInterval<T>>
+		implements DistanceTransform, Contingent {
+
+	@Parameter
+	private ThreadService ts;
+
+	@Parameter(required = false)
+	private StructuringElement element;
+
+	@SuppressWarnings("rawtypes")
+	private UnaryFunctionOp<FinalInterval, RandomAccessibleInterval> createOp;
+
+	private ExecutorService es;
+
+	@Override
+	public boolean conforms() {
+		return in().numDimensions() > 3;
+	}
+
+	@Override
+	public void initialize() {
+		es = ts.getExecutorService();
+		createOp = Functions.unary(ops(), CreateImgFromDimsAndType.class, RandomAccessibleInterval.class,
+				new FinalInterval(in()), new FloatType());
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public RandomAccessibleInterval<T> createOutput(final RandomAccessibleInterval<B> in) {
+		return createOp.compute1(new FinalInterval(in));
+	}
+
+	/*
+	 * all values are stored in a 1D array this function computes the index in
+	 * the array for a multidimensional position
+	 */
+	public int getIndex(final int[] positions, final int[] dimensSizes) {
+		int index = 0;
+		for (int i = 0; i < positions.length; i++) {
+			int temp = 1;
+			for (int j = 0; j < i; j++)
+				temp *= dimensSizes[j];
+			index += temp * positions[i];
+		}
+		return index;
+	}
+
+	/*
+	 * extended version of meijsters raster scan alogrithm to compute n-d inputs
+	 * Source: http://fab.cba.mit.edu/classes/S62.12/docs/Meijster_distance.pdf
+	 */
+	@Override
+	public void compute1(final RandomAccessibleInterval<B> in, final RandomAccessibleInterval<T> out) {
+		// stores the size of each dimension
+		final int[] dimensSizes = new int[in.numDimensions()];
+
+		// stores the actual position in the image
+		final int[] positions = new int[in.numDimensions()];
+
+		// calculates the number of points in the n-d space
+		int numPoints = 1;
+		for (int i = 0; i < in.numDimensions(); i++) {
+			numPoints *= in.dimension(i);
+			dimensSizes[i] = (int) in.dimension(i);
+			positions[i] = 0;
+		}
+		// stores the values calculated after each phase
+		final int[] actualValues = new int[numPoints];
```

Comment:
it depends on whether we need to be able to handle images with more than 2 billions pixel/voxel